modules/fundamental/src/purl/service/mod.rs

```diff
@@ -339,6 +344,100 @@ impl PurlService {
 
         Ok(res.rows_affected())
     }
+
+    #[instrument(skip(self, connection), err)]
+    pub async fn recommend_purls<C: ConnectionTrait>(
+        &self,
+        purls: &[Purl],
+        connection: &C,
+    ) -> Result<HashMap<String, Vec<RecommendEntry>>, Error> {
+        let mut recommendations = HashMap::new();
+
+        let Ok(pattern) = Regex::new("redhat-[0-9]+$") else {
+            return Err(Error::Internal(
+                "failed to compile trusted content regex".into(),
+            ));
+        };
+
+        for purl in purls {
+            let query = match purl.to_string().split_once('@') {
+                Some((p, _)) => format!("purl~{p}"),
+                None => format!("purl~{purl}"),
+            };
+            let summaries = self
+                .purls(q(&query), Default::default(), connection)
+                .await?;
+
+            let Some(ref input_version_str) = purl.version else {
+                return Err(Error::BadRequest(format!(
+                    "purl {} is missing version",
+                    purl
+                )));
+            };
+            let input_version = match lenient_semver::parse(input_version_str) {
+                Ok(v) => v,
+                Err(e) => {
+                    return Err(Error::BadRequest(format!(
+                        "version {} in purl {} is not valid semver: {}",
+                        input_version_str, purl, e
+                    )));
+                }
+            };
+            let mut recommended_purls = Vec::new();
+            for summary in summaries.items {
+                let Some(ref version) = summary.head.purl.version else {
+                    continue;
+                };
+
+                if pattern.find(version).is_none() {
+                    continue;
+                }
+
+                let mut similar_version = match lenient_semver::parse(version) {
+                    Ok(v) => v,
+                    Err(e) => {
+                        return Err(Error::Internal(format!(
+                            "{} in similar purl {} is not valid semver: {}",
+                            version, summary.head.purl, e
+                        )));
+                    }
+                };
+                // From the original version in trustification v1:
+                // remove the RHT stupid renaming because semver thinks it means pre-release
+                // and that breaks stupid comparisions.
```

Comment:
This confused me at first. I might include a link instead...

```suggestion
                // See https://github.com/trustification/trustification/blob/1d65f2c1cce303a89f1f9ca1e8cd6285f1d23de0/exhort/api/src/server.rs#L204-L205
```